app/main/views/suppliers.py

```diff
@@ -408,13 +409,16 @@ def update_supplier_framework_details(supplier_id, framework_slug):
     if not interest_record:
         abort(404, "supplier_id '{}' has not registered interest in {}".format(supplier_id, framework_slug))
 
+    uniform_now = datetime.utcnow()
+
     if 'onFramework' in update_json:
         interest_record.on_framework = update_json['onFramework']
-    if 'agreementReturned' in update_json:
-        if update_json['agreementReturned']:
-            interest_record.agreement_returned_at = datetime.utcnow()
-        else:
-            interest_record.agreement_returned_at = None
+    if update_json.get('agreementReturned'):
+        interest_record.agreement_returned_at = uniform_now
+    if update_json.get('countersigned'):
+        interest_record.countersigned_at = uniform_now
+    if 'signerDetails' in update_json:
+        interest_record.signer_details = update_json["signerDetails"]
```

Comment:
Thinking forward, we are going to need to add data to the `signerDetails` in future - when CCS do the countersigning.  So it might be better if this updated the existing signer details, rather than overwriting them completely, eg something like 
`interest_record.signer_details.update(update_json["signerDetails"])`

Would be nice to have a test around this too - i.e. a second call to the endpoint updating some existing details, and not losing any keys.